dedupe/core.py

```diff
@@ -77,7 +77,7 @@ def __call__(self, records_queue, score_queue) :
 
             try :
                 filtered_pairs = self.fieldDistance(record_pairs)
-                if filtered_pairs is not None :
+                if filtered_pairs is not None and filtered_pairs.size != 0 :
```

Comment:
I think I would prefer this to be handled by `self.fieldDistance` 

do a check of of `mask.any()` and return None if mask.any() is not true.